lib/rust/prelude/src/string.rs

```diff
@@ -214,6 +214,56 @@ impl PartialEq<ImString> for String {
     }
 }
 
+
+
+// ==================
+// === ToImString ===
+// ==================
+
+/// Conversion of a value to [`ImString`].
+#[allow(missing_docs)]
+pub trait ToImString {
```

Comment:
Is there some trait-resolution problem with using `impl From<T> for ImString`?

Reply:
It's just like `ToString`, it allows us to use method `.to_im_string`, which is sometimes just nicer. Does it make sense?

Reply:
Yeah, that makes sense.